File: windows/BlueLink.App/Domain/AutoConnectionPolicy.cs
namespace BlueLink.Domain;

public static class AutoConnectionPolicy
{
    public static bool ShouldConnect(bool trusted, bool connectedThisRun, bool manuallyDisconnected,
        bool autoConnect, bool reconnect) => trusted && !manuallyDisconnected && (connectedThisRun ? reconnect : autoConnect);
}

File: android/app/src/main/java/com/bluelink/android/domain/AutoConnectionPolicy.kt
package com.bluelink.android.domain

/** Startup discovery and reconnection after a live session are separate user choices. */
object AutoConnectionPolicy {
    fun shouldConnect(trusted: Boolean, connectedThisRun: Boolean, manuallyDisconnected: Boolean,
                      autoConnectTrusted: Boolean, reconnectAfterDisconnect: Boolean): Boolean {
        if (!trusted || manuallyDisconnected) return false
        return if (connectedThisRun) reconnectAfterDisconnect else autoConnectTrusted
    }
}
